alacritty/src/input.rs

```diff
@@ -156,12 +160,20 @@ impl<T: EventListener> Execute<T> for Action {
                 let text = ctx.clipboard_mut().load(ClipboardType::Selection);
                 paste(ctx, &text);
             },
-            Action::Command(ref program) => {
-                let args = program.args();
-                let program = program.program();
+            Action::Command(ref cmd) => {
+                let args = cmd.args().to_owned();
+                let input = cmd.input();
+                let program = cmd.program().to_string();
                 trace!("Running command {} with args {:?}", program, args);
 
-                start_daemon(program, args);
+                let input_str = input.map(|i| match i {
+                    CommandInput::VisibleText => ctx.to_string_only_visible(),
+                    CommandInput::AllText => ctx.to_string(),
+                });
+
+                spawn(move || {
+                    start_daemon(&program, &args, input_str.as_deref());
+                });
```

Comment:
Threads should ideally be spawned with a name and functions shouldn't be directly imported.

Reply:
Fixed it